seeq/addons/correlation/_seeq_worksheet_writer.py

```diff
@@ -214,7 +214,13 @@ def worksheet_corrs_and_time_shifts(signal_pairs_ids: list, workbook_id: str,
                             condition_id,
                             overwrite,
                             api_client)
-    return "/".join([spy.client.host.replace('/api', ""), 'workbook', workbook_id, 'worksheet', existing_worksheet.id])
+
+    try:
+        seeq_url_ = spy.session.public_url
```

Comment:
I would use an if-statement instead of the try/except block since that exception is very broad. 
 Something like: `if hasattr(spy.session, 'public_url'):`

Additionally, we need to try getting the URL from `spy.session.private_url` if the `public_url` is either null or the property doesn't exist. However, `spy.session.private_url` can be an IP address that will be unreachable. Thus, you can check with a regex whether the value in `spy.session.private_url` is an IP (i.e. has the format XXX.XXX.XXX.XXX).

After the above two checks (`public_url` and `private_url`) I don't know if we still want to go to `spy.client.host` since that property comes from the ApiClient directly

Reply:
agreed that is a good path forward, just tested `public_url `and `private_url` and they both as in R52 and above which means we can completely forgo `spy.client.host`